packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -490,17 +522,14 @@ class Auth {
                     !!refreshTokenGuest
                 )
             } catch (error) {
-                // If the refresh token is invalid, we need to re-login the user
-                if (error instanceof Error && 'response' in error) {
-                    // commerce-sdk-isomorphic throws a `ResponseError`, but doesn't export the class.
-                    // We can't use `instanceof`, so instead we just check for the `response` property
-                    // and assume it is a fetch Response.
-                    const json = await (error['response'] as Response).json()
-                    if (json.message === 'invalid refresh_token') {
-                        // clean up storage and restart the login flow
-                        this.clearStorage()
-                    }
-                }
+                // If the refresh token login fails, we fall back to the new guest refresh login
+                const {status_code, responseMessage} = await this.extractResponseError(
```

Comment:
Dumb question, why this is async function? it is an pure function to me

Reply:
Because it needs to consume the response body and calling `json()` is an async operation.

Reply:
ah gotcha. 👍